homeassistant/components/verisure/coordinator.py

```diff
@@ -56,60 +62,115 @@ async def async_login(self) -> bool:
 
     async def async_logout(self, _event: Event) -> None:
         """Logout from Verisure."""
-        try:
-            await self.hass.async_add_executor_job(self.verisure.logout)
-        except VerisureError as ex:
-            LOGGER.error("Could not log out from verisure, %s", ex)
 
     async def _async_update_data(self) -> dict:
         """Fetch data from Verisure."""
+        try:
+            await self.hass.async_add_executor_job(self.verisure.update_cookie)
+        except (VerisureLoginError, VerisureResponseError) as ex:
+            LOGGER.error("Credentials expired for Verisure, %s", ex)
+            raise ConfigEntryAuthFailed("Credentials expired for Verisure") from ex
         try:
             overview = await self.hass.async_add_executor_job(
-                self.verisure.get_overview
+                self.verisure.request,
+                self.verisure.arm_state(),
+                self.verisure.broadband(),
+                self.verisure.cameras(),
+                self.verisure.climate(),
+                self.verisure.door_window(),
+                self.verisure.smart_lock(),
+                self.verisure.smartplugs(),
             )
-        except VerisureResponseError as ex:
-            LOGGER.error("Could not read overview, %s", ex)
-            if ex.status_code == HTTPStatus.SERVICE_UNAVAILABLE:
-                LOGGER.info("Trying to log in again")
-                await self.async_login()
-                return {}
-            raise
+        except VerisureResponseError as err:
+            LOGGER.debug("Cookie expired or service unavailable, %s", err)
+            overview = self._overview
+            try:
+                await self.hass.async_add_executor_job(self.verisure.update_cookie)
+            except VerisureResponseError as ex:
+                raise ConfigEntryAuthFailed("Credentials for Verisure expired.") from ex
+            except Exception as ex:
+                LOGGER.error("Could not read overview, %s", ex)
+                raise UpdateFailed("Could not read overview") from ex
+        except VerisureError as err:
+            LOGGER.error("Could not read overview, %s", err)
+            raise UpdateFailed("Could not read overview") from err
+
+        def unpack(overview: list, value: str) -> dict | list:
+            unpacked = [
+                item["data"]["installation"][value]
+                for item in overview
+                if value in item.get("data", {}).get("installation", {})
```

Comment:
As pointed out in the previous PR, an account can hold multiple installations. The config entry, however, is tied into a single installation.

Only the data matching the selected installation should be gathered.

Reply:
The installation that is read from it selected with this statement, the results "should" only contain information about the configured installation.

```python
await self.hass.async_add_executor_job(
            self.verisure.set_giid, self.entry.data[CONF_GIID]
        ) 
```

Reply:
It would be better to set_giid synchronously btw

Reply:
> It would be better to set_giid synchronously btw

How?

Reply:
Are we in agreement that this comment is resolved because the GIID is set at login?

Reply:
Not fully, if we can expect the installation to be there, we don't need to use `get()`

Reply:
Dict key `installation` is not the same as the giid. Are we still talking about the same thing?